### PR TITLE
Install secrets and hooks on FreeBSD buildkite-worker images

### PR DESCRIPTION
**Pull request by ararslan**

### CODE REVIEW

**Comment by DilumAluthge** [comment]
Do we actually need any secrets on the FreeBSD agents? For example, there's no code signing done on FreeBSD, right?

**Comment by ararslan** [comment]
Right now I'm only taking the agent key pair and API token, the same as Windows.

File: freebsd-kvm/buildkite-worker/kvm_machine.pkr.hcl.template [hunk 1]
```diff
@@ -46,6 +46,16 @@ source "qemu" "freebsd12" {
...
 
+    provisioner "file" {
+        source = "../../secrets"
+        destination = "/tmp/secrets"
+    }
+
+    provisioner "file" {
+        source = "../../hooks"
+        destination = "/tmp/hooks"
+    }
```
Comment (ararslan) [inline freebsd-kvm/buildkite-worker/kvm_machine.pkr.hcl.template:57]: FWIW based on my read of the Packer documentation, this should be equivalent but I'm not entirely sure.
```suggestion
    provisioner "file" {
        sources = [
            "../../secrets",
            "../../hooks",
        ]
        destination = "/tmp/"
    }
```

**Comment by DilumAluthge** [comment]
Hmm, but why do we even need the agent keypair and the API token? AFAICT, we only need those so that we can decrypt other secrets, such as the codesigning secrets on Windows and macOS. If there aren't any other secrets that we need to decrypt on FreeBSD, then I don't think we need the agent keypair or API token.

**Comment by ararslan** [comment]
I was just going by what Elliot told me to do:

> Okay, another problem with the freebsd agent right now is that we’re missing all the hooks and secrets and whatnot

So I assumed it just needed whatever secrets Windows is using. But maybe all it's actually missing is the hooks.

File: freebsd-kvm/buildkite-worker/setup_scripts/install-cryptic-secrets.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+
+set -e
+
+echo "-> Installing jq"
+# Needed for the secrets cleanup hook
```
Comment (staticfloat) [inline freebsd-kvm/buildkite-worker/setup_scripts/install-cryptic-secrets.sh:6]:
```suggestion
# Needed to parse buildkite API responses
```

It's actually `shred` that's needed for secrets cleanup. ;)

**Comment by DilumAluthge** [comment]
@staticfloat What secrets do we need on FreeBSD?

**Comment by ararslan** [comment]
He can't say, it's a secret

**Comment by ararslan** [comment]
Based on https://buildkite.com/julialang/julia-buildkite/builds/1200#0184f949-68b9-44db-b749-d763e0dedbe1 it looks like at least the agent secret key

**Comment by ararslan** [comment]
And it appears Elliot deployed this change since the FreeBSD build in https://github.com/JuliaCI/julia-buildkite/pull/266 is now working! Should we merge this then?

**Pull request merged**